### ISSUE TITLE
ID Generation

### ISSUE THREAD

**Issue by bgreenawald** [bgreenawald/madgab-online#35]
Does the frontend enforce the uniqueness of game IDs? I was hoping to change the process for generating game ID's, replacing the _get_names_ endpoint with a _check_name_ endpoint (the client being able to access the name of all active games seems like a great way for people to sabotage other peoples games). Merely having an endpoint that returns True or False if a given name is already in use would be much more secure.

That being said, when looking at the frontend code, I can't find where the client checks to ensure that the new ID isn't currently in use.

**Comment by JocelynYH** [comment]
This is what we have on the frontend:
https://github.com/bgreenawald/madgab-online/blob/6f8a72818c2fb2aaaddc4f756d5ef8dd15d208be/frontend/src/store/actions.js#L22

**Comment by JocelynYH** [comment]
The boolean endpoint is a great idea.

**Comment by bgreenawald** [comment]
I've changed the endpoints, deprecating the _/api/get_names_ endpoint and replacing it with _/api/check_name/\<ID\>_ which return a payload with structure
```
{
    "in_use": True|False
}
```

**Comment by bgreenawald** [comment]
And maybe I'm misunderstanding the frontend code, but I'm still not sure where the client blocks ID suggestions to users that are already in use. We obviously can't prevent a user from going to an ID that is already in use and crashing another user's game, but having the client use that new endpoint to give users a warning that an ID is already in use and prompting them to try a different one would be good.

**Comment by JocelynYH** [comment]
Yeah, the frontend code just generates another ID if the random number generator landed on an ID already in use.  It's that function you wrote - I didn't touch it really. 

But yes, we don't actually have anything that'd prevent the client from deciding to copy an in-use ID.  

So, from the frontend, this is what we wanna do yes?  user submits an ID (provided or auto generated), then frontend hits the new API - if `in_use === false`, happy path, else show error?

**Comment by bgreenawald** [comment]
That's exactly what I'm thinking!

**Comment by JocelynYH** [comment]
Perfect!  I'll do that then!
